lib/ansible/modules/cloud/vmware/vmware_tag_manager.py

```diff
@@ -218,15 +236,24 @@ def ensure_state(self):
         removed_tags_for_set = False
         for tag in self.tag_names:
             category_obj, category_name, tag_name = None, None, None
-            if ":" in tag:
-                # User specified category
-                category_name, tag_name = tag.split(":", 1)
-                category_obj = self.search_svc_object_by_name(self.category_service, category_name)
-                if not category_obj:
-                    self.module.fail_json(msg="Unable to find the category %s" % category_name)
-            else:
-                # User specified only tag
-                tag_name = tag
+            if isinstance(tag, dict):
```

Comment:
could create a function that take a tag_name entry (so, dict, or string). and return `(category_obj, tag_obj)`. This way we can simplify the loop. Basically line 239 to 265 should move in the function.